resources/services/s3/buckets.go

```diff
@@ -787,3 +799,41 @@ func resolveBucketTagging(ctx context.Context, meta schema.ClientMeta, resource
 	}
 	return resource.Set("tags", tags)
 }
+
+func resolveBucketOwnershipControls(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, bucketName, bucketRegion string) error {
+	c := meta.(*client.Client)
+	svc := c.Services().S3
+
+	getBucketOwnershipControlOutput, err := svc.GetBucketOwnershipControls(ctx, &s3.GetBucketOwnershipControlsInput{Bucket: aws.String(bucketName)}, func(options *s3.Options) {
+		options.Region = bucketRegion
+	})
+
+	if err != nil {
+		if c.IsNotFoundError(err) { // Not all buckets have ownership controls
+			return nil
+		}
+
+		if client.IgnoreAccessDeniedServiceDisabled(err) {
+			meta.Logger().Warn("received access denied on GetBucketTagging", "bucket", bucketName, "err", err)
+			return nil
+		}
+
+		return err
+	}
+
+	if getBucketOwnershipControlOutput == nil {
+		return nil
+	}
+
+	ownershipControlRules := getBucketOwnershipControlOutput.OwnershipControls.Rules
+
+	if len(ownershipControlRules) == 0 {
+		return nil
+	}
+
+	if len(ownershipControlRules) > 1 {
+		return errors.New("more than one ownership control rules")
+	}
```

Comment:
Why can we not support more than 1 rule?

Reply:
Agreed. Let's change it to a text-array